core/src/mindustry/world/blocks/production/Drill.java

```diff
@@ -403,6 +412,21 @@ public void read(Reads read, byte revision){
                 warmup = read.f();
             }
         }
-    }
+        @Override
+        public float warmup(){
+            return warmup;
+        }
+
+        @Override
+        public float progress() {
+            float drillProgress = (dominantItem == null) ? 0f : Mathf.clamp(progress / getDrillTime(dominantItem));
+            float consumeProgress = Mathf.clamp(consTimer / consumeTime);
 
+            return (drillProgress + consumeProgress) / 2f;
```

Comment:
Why is consumeProgress even a factor here? It has nothing to do with progress until the next item. Why is it averaged?